src/RemoteAPI/related/class-related-caching-decorator.php

```diff
@@ -42,10 +41,10 @@ class Related_Caching_Decorator implements Proxy {
 	/**
 	 * Constructor.
 	 *
-	 * @param Proxy           $proxy The Proxy object to cache.
-	 * @param WP_Object_Cache $cache The WordPress object cache instance.
+	 * @param Proxy         $proxy The Proxy object to cache.
+	 * @param Cache_Adapter $cache An object cache instance.
 	 */
-	public function __construct( Proxy $proxy, WP_Object_Cache $cache ) {
+	public function __construct( Proxy $proxy, Cache_Adapter $cache ) {
```

Comment:
```suggestion
	public function __construct( Proxy $proxy, Cache $cache ) {
```

Cache_Adapter is the class, but the interface is what should be type declared here.

Reply:
The doc on this constructor, and `$cache` class property doc will also need updating.